src/context.cpp

```diff
@@ -422,41 +423,77 @@ namespace librealsense
-            _index(index)
+            _name(name)
         {
         }
 
         void open( const stream_profiles & profiles ) override
         {
-            std::vector< rs2_video_stream > rs2_profs;
+            std::vector< realdds::dds_video_stream::profile > video_profiles;
+            std::vector< realdds::dds_motion_stream::profile > motion_profiles;
             for (size_t i = 0; i < profiles.size(); ++i)
             {
-                rs2_profs[i].fps = profiles[i]->get_framerate();
-                rs2_profs[i].fmt = profiles[i]->get_format();
-                rs2_profs[i].type = profiles[i]->get_stream_type();
-                const auto&& vsp = As<video_stream_profile, stream_profile_interface>( profiles[i] );
-                rs2_profs[i].width = vsp ? vsp->get_width() : 0;
-                rs2_profs[i].height = vsp ? vsp->get_height() : 0;
+                if ( Is<video_stream_profile, stream_profile_interface>( profiles[i] ) )
+                {
+                    realdds::dds_video_stream::profile video_profile;
+                    video_profile.uid       = profiles[i]->get_unique_id();
+                    video_profile.framerate = profiles[i]->get_framerate();
+                    video_profile.type      = profiles[i]->get_stream_type();
+                    video_profile.format    = profiles[i]->get_format();
+                    video_profile.index     = profiles[i]->get_stream_index();
+                    const auto && vsp       = As<video_stream_profile, stream_profile_interface>( profiles[i] );
+                    video_profile.width     = vsp->get_width();
+                    video_profile.height    = vsp->get_height();
+
+                    video_profiles.push_back( video_profile );
+                }
+                else if ( Is<motion_stream_profile, stream_profile_interface>( profiles[i] ) )
+                {
+                    realdds::dds_motion_stream::profile motion_profile;
+                    motion_profile.uid       = profiles[i]->get_unique_id();
+                    motion_profile.framerate = profiles[i]->get_framerate();
+                    motion_profile.format    = profiles[i]->get_format();
+                    motion_profile.type      = profiles[i]->get_stream_type();
+                    motion_profile.index     = profiles[i]->get_stream_index();
+
+                    motion_profiles.push_back( motion_profile );
+                }
+
+            }
+
+            if ( video_profiles.size() > 0 )
+            {
+                _dev->open( video_profiles );
+            }
+            if ( motion_profiles.size() > 0 )
+            {
+                _dev->open( motion_profiles );
             }
-            _dev->sensor_open( _index, rs2_profs );
+
             software_sensor::open( profiles );
         }
 
         void close() override
         {
-            _dev->sensor_close( _index );
+            std::vector< std::pair< int16_t, int8_t > > profiles_to_close;
```

Comment:
Can you add in a comment the pair fields?

Reply:
Done